### PR TITLE
Handle diff not encoded as utf-8 and add ability to disable diff computation when managing a repository

### PR DESCRIPTION
**Pull request by Nikokrock**

### CODE REVIEW

File: e3/text.py [hunk 1]
```diff
@@ -20,3 +20,26 @@ def format_with_dict(pattern, values):
     key_regexp = r"|".join([r'\(%s\)' % k for k in values])
     result = re.sub(r'%%(?!%s)' % key_regexp, r'%%', pattern)
     return result % values
+
+
+def bytes_as_str(content):
+    """Safely convert bytes in a python string.
```
Comment (enzbang) [inline e3/text.py]: to

File: e3/text.py [hunk 2]
```diff
@@ -20,3 +20,26 @@ def format_with_dict(pattern, values):
...
+
+    The conversion function attempt to convert to decode utf-8 and
+    in case of decoding error fallback on a safe representation in
```
Comment (enzbang) [inline e3/text.py]: fall back

**Comment by brobecke** [comment]
The changes themselves look reasonable to me, but I don't understand why we are not
adding some tests for these. Is it because we are pressed for time? Or because we feel
this isn't necessary in this case. I would agree with going fast if it is pressing (but with
a note and a TN to finish this later), but I would argue against the "not worth it" reason.

File: e3/text.py [hunk 3]
```diff
@@ -20,3 +20,26 @@ def format_with_dict(pattern, values):
     key_regexp = r"|".join([r'\(%s\)' % k for k in values])
     result = re.sub(r'%%(?!%s)' % key_regexp, r'%%', pattern)
     return result % values
+
+
+def bytes_as_str(content):
+    """Safely convert bytes to a python string.
+
+    The conversion function attempt to convert to decode utf-8 and
```
Comment (brobecke) [inline e3/text.py]: A bit hard to process, at least for the first line... How about?

This function attempts the conversion assuming a utf-8 enconding.
If this triggers a conversion error, it then falls back to a safe representation,
where some characters might be escaped.

**Pull request merged**